app/upgrades/signetlaunch/upgrades.go

```diff
@@ -79,6 +101,115 @@ func LoadBTCHeadersFromData() ([]*btclighttypes.BTCHeaderInfo, error) {
 	return gs.BtcHeaders, nil
 }
 
+// LoadSignedFPsFromData returns the finality providers from the json string.
+// It also verifies if the msg is correctly signed and is valid to be inserted.
+func LoadSignedFPsFromData(cdc codec.Codec, txJSONDecoder sdk.TxDecoder) ([]*btcstktypes.MsgCreateFinalityProvider, error) {
+	buff := bytes.NewBufferString(SignedFPsStr)
+
+	var d dataSignedFps
+	err := json.Unmarshal(buff.Bytes(), &d)
+	if err != nil {
+		return nil, err
+	}
+
+	fps := make([]*btcstktypes.MsgCreateFinalityProvider, len(d.SignedTxsFP))
+	for i, txAny := range d.SignedTxsFP {
+		txBytes, err := json.Marshal(txAny)
+		if err != nil {
+			return nil, err
+		}
+
+		tx, err := txJSONDecoder(txBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		fp, err := parseCreateFPFromSignedTx(cdc, tx)
+		if err != nil {
+			return nil, err
+		}
+
+		fps[i] = fp
+	}
+
+	// sorts all the FPs by their addresses
+	sort.Slice(fps, func(i, j int) bool {
+		return fps[i].Addr > fps[j].Addr
+	})
+
+	return fps, nil
+}
+
+func parseCreateFPFromSignedTx(cdc codec.Codec, tx sdk.Tx) (*btcstktypes.MsgCreateFinalityProvider, error) {
+	msgsV2, err := tx.GetMsgsV2()
+	if err != nil {
+		return nil, err
+	}
+
+	msgs := tx.GetMsgs()
+	if len(msgs) == 0 {
+		return nil, fmt.Errorf("invalid msg, there is no msg inside the tx %+v", tx)
+	}
+
+	// each tx should only contain one msg
```

Comment:
why? we should explain here

Reply:
it is what we expect, for each signer to only submit one message per transaction 
this is the format we are going to expect the finality providers submit to us